src/storage/context.ts

```diff
@@ -411,12 +426,18 @@ export class StorageContext {
       )
     }
 
+    // Convert options to metadata format - merge withCDN flag into metadata if needed
+    const baseMetadata = options.metadata ?? []
+    const requestedMetadata = hasWithCDN(baseMetadata)
+      ? baseMetadata
+      : [...baseMetadata, ...withCDNToMetadata(options.withCDN ?? false)]
```

Comment:
it's weird to concat an array with an array that has 0 or 1 items. Can this just be append, like
```
if (options.withCDN && !hasWithCDN(baseMetadata)) {
    baseMetadata.push(withCDNMetadataItem())
}
```

Reply:
Could be but we end up in roughly the same place except that we've also now modified the user's `options.metadata` - so we need splat that anyway to get a clone we can mess with, (or could `slice`, same thing), the splat of a 0||1 length array ends up being the same as `if(){push}`, just a terse form of it. Possibly the potential saving here would be if we _know_ we don't need to modify the input `options.metadata` and the user has provided one.